x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/InferenceIngestIT.java

```diff
@@ -94,13 +112,29 @@ public void testPathologicalPipelineCreationAndDeletion() throws Exception {
                     QueryBuilders.existsQuery("ml.inference.classification.predicted_value"))));
 
         assertThat(EntityUtils.toString(searchResponse.getEntity()), containsString("\"value\":10"));
+        assertBusy(() -> {
+            try {
+                Response statsResponse = client().performRequest(new Request("GET",
+                    "_ml/inference/" + classificationModelId + "/_stats"));
+                assertThat(EntityUtils.toString(statsResponse.getEntity()), containsString("\"inference_count\":10"));
+                statsResponse = client().performRequest(new Request("GET", "_ml/inference/" + regressionModelId + "/_stats"));
+                assertThat(EntityUtils.toString(statsResponse.getEntity()), containsString("\"inference_count\":10"));
+            } catch (ResponseException ex) {
+                //this could just mean shard failures.
```

Comment:
Shouldn't this "catch" clause call "Assert.fail()"?
If it doesn't I think the whole block won't be retried when the `ResponseException` happens...